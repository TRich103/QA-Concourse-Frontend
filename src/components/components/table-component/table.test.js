import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme,{shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import QATable from './qa-table.component';
import ReactTable from "react-table";
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()})

const tableData = {
                      Headers: ['Name', 'Age'],
                      Rows: [{'Name': 'John', 'Age': '23'},
                             {'Name': 'Jane', 'Age': '26'}]
                  }

describe('Table component', () =>{

  //Creates snapshot of desired table
  test('renders', () => {
    const table = shallow(      
      <ReactTable
        className="-striped -highlight"
        data= {[{'Name': 'John', 'Age': '23'},
                {'Name': 'Jane', 'Age': '26'}]}
        columns={[{'Header': 'Name', 'accessor': 'Name'}, {'Header': 'Age', 'accessor': 'Age'}]}
        showPagination={false}
        style={{
                height: "400px"
        }}
    />
  )
    expect(toJson(table)).toMatchSnapshot()
  })

  //Checks if QATable component matches desired table
  test('Populates data from props', () => {
    const table = shallow(<QATable data={tableData}/>);
    expect(toJson(table)).toMatchSnapshot()
  })

  //Checks if setting props updates table
  test('Updates table on prop change', () => {
    const blankTable = shallow(<QATable data={{Headers:[], Rows:[]}}/>);
    const filledTable = blankTable.setProps({data: tableData})
    expect(filledTable).toMatchSnapshot()
  })
})