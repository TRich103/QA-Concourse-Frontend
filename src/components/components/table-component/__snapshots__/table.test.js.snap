// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component Populates data from props 1`] = `
<ReactTable
  AggregatedComponent={[Function]}
  ExpanderComponent={[Function]}
  FilterComponent={[Function]}
  LoadingComponent={[Function]}
  NoDataComponent={[Function]}
  PadRowComponent={[Function]}
  PaginationComponent={[Function]}
  PivotValueComponent={[Function]}
  ResizerComponent={[Function]}
  TableComponent={[Function]}
  TbodyComponent={[Function]}
  TdComponent={[Function]}
  TfootComponent={[Function]}
  ThComponent={[Function]}
  TheadComponent={[Function]}
  TrComponent={[Function]}
  TrGroupComponent={[Function]}
  aggregatedKey="_aggregated"
  className="-striped -highlight"
  collapseOnDataChange={true}
  collapseOnPageChange={true}
  collapseOnSortingChange={true}
  column={
    Object {
      "Aggregated": undefined,
      "Cell": undefined,
      "Expander": undefined,
      "Filter": undefined,
      "Footer": undefined,
      "Header": undefined,
      "Pivot": undefined,
      "PivotValue": undefined,
      "Placeholder": undefined,
      "aggregate": undefined,
      "className": "",
      "filterAll": false,
      "filterMethod": undefined,
      "filterable": undefined,
      "footerClassName": "",
      "footerStyle": Object {},
      "getFooterProps": [Function],
      "getHeaderProps": [Function],
      "getProps": [Function],
      "headerClassName": "",
      "headerStyle": Object {},
      "minResizeWidth": 11,
      "minWidth": 100,
      "resizable": undefined,
      "show": true,
      "sortMethod": undefined,
      "sortable": undefined,
      "style": Object {},
    }
  }
  columns={
    Array [
      Object {
        "Header": undefined,
        "accessor": undefined,
        "width": undefined,
      },
      Object {
        "Header": undefined,
        "accessor": undefined,
        "width": undefined,
      },
    ]
  }
  data={
    Array [
      Object {
        "Age": "23",
        "Name": "John",
      },
      Object {
        "Age": "26",
        "Name": "Jane",
      },
    ]
  }
  defaultExpanded={Object {}}
  defaultFilterMethod={[Function]}
  defaultFiltered={Array []}
  defaultPage={0}
  defaultPageSize={20}
  defaultResized={Array []}
  defaultSortDesc={false}
  defaultSortMethod={[Function]}
  defaultSorted={Array []}
  expanderDefaults={
    Object {
      "filterable": false,
      "resizable": false,
      "sortable": false,
      "width": 35,
    }
  }
  filterable={false}
  freezeWhenExpanded={false}
  getLoadingProps={[Function]}
  getNoDataProps={[Function]}
  getPaginationProps={[Function]}
  getProps={[Function]}
  getResizerProps={[Function]}
  getTableProps={[Function]}
  getTbodyProps={[Function]}
  getTdProps={[Function]}
  getTfootProps={[Function]}
  getTfootTdProps={[Function]}
  getTfootTrProps={[Function]}
  getTheadFilterProps={[Function]}
  getTheadFilterThProps={[Function]}
  getTheadFilterTrProps={[Function]}
  getTheadGroupProps={[Function]}
  getTheadGroupThProps={[Function]}
  getTheadGroupTrProps={[Function]}
  getTheadProps={[Function]}
  getTheadThProps={[Function]}
  getTheadTrProps={[Function]}
  getTrGroupProps={[Function]}
  getTrProps={[Function]}
  groupedByPivotKey="_groupedByPivot"
  indexKey="_index"
  loading={false}
  loadingText="Loading..."
  multiSort={true}
  nestingLevelKey="_nestingLevel"
  nextText="Next"
  noDataText="No rows found"
  ofText="of"
  onFetchData={[Function]}
  originalKey="_original"
  pageJumpText="jump to page"
  pageSizeOptions={
    Array [
      5,
      10,
      20,
      25,
      50,
      100,
    ]
  }
  pageText="Page"
  pivotDefaults={Object {}}
  pivotIDKey="_pivotID"
  pivotValKey="_pivotVal"
  previousText="Previous"
  resizable={true}
  resolveData={[Function]}
  rowsSelectorText="rows per page"
  rowsText="rows"
  showPageJump={true}
  showPageSizeOptions={true}
  showPagination={false}
  showPaginationBottom={true}
  showPaginationTop={false}
  sortable={true}
  style={
    Object {
      "height": "400px",
    }
  }
  subRowsKey="_subRows"
/>
`;

exports[`Table component Updates table on prop change 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QATable
    data={
      Object {
        "Headers": Array [
          "Name",
          "Age",
        ],
        "Rows": Array [
          Object {
            "Age": "23",
            "Name": "John",
          },
          Object {
            "Age": "26",
            "Name": "Jane",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "AggregatedComponent": [Function],
      "ExpanderComponent": [Function],
      "FilterComponent": [Function],
      "LoadingComponent": [Function],
      "NextComponent": undefined,
      "NoDataComponent": [Function],
      "PadRowComponent": [Function],
      "PaginationComponent": [Function],
      "PivotComponent": undefined,
      "PivotValueComponent": [Function],
      "PreviousComponent": undefined,
      "ResizerComponent": [Function],
      "TableComponent": [Function],
      "TbodyComponent": [Function],
      "TdComponent": [Function],
      "TfootComponent": [Function],
      "ThComponent": [Function],
      "TheadComponent": [Function],
      "TrComponent": [Function],
      "TrGroupComponent": [Function],
      "aggregatedKey": "_aggregated",
      "className": "-striped -highlight",
      "collapseOnDataChange": true,
      "collapseOnPageChange": true,
      "collapseOnSortingChange": true,
      "column": Object {
        "Aggregated": undefined,
        "Cell": undefined,
        "Expander": undefined,
        "Filter": undefined,
        "Footer": undefined,
        "Header": undefined,
        "Pivot": undefined,
        "PivotValue": undefined,
        "Placeholder": undefined,
        "aggregate": undefined,
        "className": "",
        "filterAll": false,
        "filterMethod": undefined,
        "filterable": undefined,
        "footerClassName": "",
        "footerStyle": Object {},
        "getFooterProps": [Function],
        "getHeaderProps": [Function],
        "getProps": [Function],
        "headerClassName": "",
        "headerStyle": Object {},
        "minResizeWidth": 11,
        "minWidth": 100,
        "resizable": undefined,
        "show": true,
        "sortMethod": undefined,
        "sortable": undefined,
        "style": Object {},
      },
      "columns": Array [
        Object {
          "Header": undefined,
          "accessor": undefined,
          "width": undefined,
        },
        Object {
          "Header": undefined,
          "accessor": undefined,
          "width": undefined,
        },
      ],
      "data": Array [
        Object {
          "Age": "23",
          "Name": "John",
        },
        Object {
          "Age": "26",
          "Name": "Jane",
        },
      ],
      "defaultExpanded": Object {},
      "defaultFilterMethod": [Function],
      "defaultFiltered": Array [],
      "defaultPage": 0,
      "defaultPageSize": 20,
      "defaultResized": Array [],
      "defaultSortDesc": false,
      "defaultSortMethod": [Function],
      "defaultSorted": Array [],
      "expanderDefaults": Object {
        "filterable": false,
        "resizable": false,
        "sortable": false,
        "width": 35,
      },
      "filterable": false,
      "freezeWhenExpanded": false,
      "getLoadingProps": [Function],
      "getNoDataProps": [Function],
      "getPaginationProps": [Function],
      "getProps": [Function],
      "getResizerProps": [Function],
      "getTableProps": [Function],
      "getTbodyProps": [Function],
      "getTdProps": [Function],
      "getTfootProps": [Function],
      "getTfootTdProps": [Function],
      "getTfootTrProps": [Function],
      "getTheadFilterProps": [Function],
      "getTheadFilterThProps": [Function],
      "getTheadFilterTrProps": [Function],
      "getTheadGroupProps": [Function],
      "getTheadGroupThProps": [Function],
      "getTheadGroupTrProps": [Function],
      "getTheadProps": [Function],
      "getTheadThProps": [Function],
      "getTheadTrProps": [Function],
      "getTrGroupProps": [Function],
      "getTrProps": [Function],
      "groupedByPivotKey": "_groupedByPivot",
      "indexKey": "_index",
      "loading": false,
      "loadingText": "Loading...",
      "multiSort": true,
      "nestingLevelKey": "_nestingLevel",
      "nextText": "Next",
      "noDataText": "No rows found",
      "ofText": "of",
      "onExpandedChange": undefined,
      "onFetchData": [Function],
      "onFilteredChange": undefined,
      "onPageChange": undefined,
      "onPageSizeChange": undefined,
      "onResizedChange": undefined,
      "onSortedChange": undefined,
      "originalKey": "_original",
      "pageJumpText": "jump to page",
      "pageSizeOptions": Array [
        5,
        10,
        20,
        25,
        50,
        100,
      ],
      "pageText": "Page",
      "pivotBy": undefined,
      "pivotDefaults": Object {},
      "pivotIDKey": "_pivotID",
      "pivotValKey": "_pivotVal",
      "previousText": "Previous",
      "resizable": true,
      "resolveData": [Function],
      "rowsSelectorText": "rows per page",
      "rowsText": "rows",
      "showPageJump": true,
      "showPageSizeOptions": true,
      "showPagination": false,
      "showPaginationBottom": true,
      "showPaginationTop": false,
      "sortable": true,
      "style": Object {
        "height": "400px",
      },
      "subRowsKey": "_subRows",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "AggregatedComponent": [Function],
        "ExpanderComponent": [Function],
        "FilterComponent": [Function],
        "LoadingComponent": [Function],
        "NextComponent": undefined,
        "NoDataComponent": [Function],
        "PadRowComponent": [Function],
        "PaginationComponent": [Function],
        "PivotComponent": undefined,
        "PivotValueComponent": [Function],
        "PreviousComponent": undefined,
        "ResizerComponent": [Function],
        "TableComponent": [Function],
        "TbodyComponent": [Function],
        "TdComponent": [Function],
        "TfootComponent": [Function],
        "ThComponent": [Function],
        "TheadComponent": [Function],
        "TrComponent": [Function],
        "TrGroupComponent": [Function],
        "aggregatedKey": "_aggregated",
        "className": "-striped -highlight",
        "collapseOnDataChange": true,
        "collapseOnPageChange": true,
        "collapseOnSortingChange": true,
        "column": Object {
          "Aggregated": undefined,
          "Cell": undefined,
          "Expander": undefined,
          "Filter": undefined,
          "Footer": undefined,
          "Header": undefined,
          "Pivot": undefined,
          "PivotValue": undefined,
          "Placeholder": undefined,
          "aggregate": undefined,
          "className": "",
          "filterAll": false,
          "filterMethod": undefined,
          "filterable": undefined,
          "footerClassName": "",
          "footerStyle": Object {},
          "getFooterProps": [Function],
          "getHeaderProps": [Function],
          "getProps": [Function],
          "headerClassName": "",
          "headerStyle": Object {},
          "minResizeWidth": 11,
          "minWidth": 100,
          "resizable": undefined,
          "show": true,
          "sortMethod": undefined,
          "sortable": undefined,
          "style": Object {},
        },
        "columns": Array [
          Object {
            "Header": undefined,
            "accessor": undefined,
            "width": undefined,
          },
          Object {
            "Header": undefined,
            "accessor": undefined,
            "width": undefined,
          },
        ],
        "data": Array [
          Object {
            "Age": "23",
            "Name": "John",
          },
          Object {
            "Age": "26",
            "Name": "Jane",
          },
        ],
        "defaultExpanded": Object {},
        "defaultFilterMethod": [Function],
        "defaultFiltered": Array [],
        "defaultPage": 0,
        "defaultPageSize": 20,
        "defaultResized": Array [],
        "defaultSortDesc": false,
        "defaultSortMethod": [Function],
        "defaultSorted": Array [],
        "expanderDefaults": Object {
          "filterable": false,
          "resizable": false,
          "sortable": false,
          "width": 35,
        },
        "filterable": false,
        "freezeWhenExpanded": false,
        "getLoadingProps": [Function],
        "getNoDataProps": [Function],
        "getPaginationProps": [Function],
        "getProps": [Function],
        "getResizerProps": [Function],
        "getTableProps": [Function],
        "getTbodyProps": [Function],
        "getTdProps": [Function],
        "getTfootProps": [Function],
        "getTfootTdProps": [Function],
        "getTfootTrProps": [Function],
        "getTheadFilterProps": [Function],
        "getTheadFilterThProps": [Function],
        "getTheadFilterTrProps": [Function],
        "getTheadGroupProps": [Function],
        "getTheadGroupThProps": [Function],
        "getTheadGroupTrProps": [Function],
        "getTheadProps": [Function],
        "getTheadThProps": [Function],
        "getTheadTrProps": [Function],
        "getTrGroupProps": [Function],
        "getTrProps": [Function],
        "groupedByPivotKey": "_groupedByPivot",
        "indexKey": "_index",
        "loading": false,
        "loadingText": "Loading...",
        "multiSort": true,
        "nestingLevelKey": "_nestingLevel",
        "nextText": "Next",
        "noDataText": "No rows found",
        "ofText": "of",
        "onExpandedChange": undefined,
        "onFetchData": [Function],
        "onFilteredChange": undefined,
        "onPageChange": undefined,
        "onPageSizeChange": undefined,
        "onResizedChange": undefined,
        "onSortedChange": undefined,
        "originalKey": "_original",
        "pageJumpText": "jump to page",
        "pageSizeOptions": Array [
          5,
          10,
          20,
          25,
          50,
          100,
        ],
        "pageText": "Page",
        "pivotBy": undefined,
        "pivotDefaults": Object {},
        "pivotIDKey": "_pivotID",
        "pivotValKey": "_pivotVal",
        "previousText": "Previous",
        "resizable": true,
        "resolveData": [Function],
        "rowsSelectorText": "rows per page",
        "rowsText": "rows",
        "showPageJump": true,
        "showPageSizeOptions": true,
        "showPagination": false,
        "showPaginationBottom": true,
        "showPaginationTop": false,
        "sortable": true,
        "style": Object {
          "height": "400px",
        },
        "subRowsKey": "_subRows",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Table component renders 1`] = `
<div
  className="ReactTable -striped -highlight"
  style={
    Object {
      "height": "400px",
    }
  }
>
  <TableComponent
    className=""
  >
    <Thead
      className="-header"
      style={
        Object {
          "minWidth": "200px",
        }
      }
    >
      <TrComponent>
        <ThComponent
          className="rt-resizable-header -cursor-pointer"
          key="0-Name"
          style={
            Object {
              "flex": "100 0 auto",
              "maxWidth": null,
              "width": "100px",
            }
          }
          toggleSort={[Function]}
        >
          <div
            className="rt-resizable-header-content"
          >
            Name
          </div>
          <Resizer
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          />
        </ThComponent>
        <ThComponent
          className="rt-resizable-header -cursor-pointer"
          key="1-Age"
          style={
            Object {
              "flex": "100 0 auto",
              "maxWidth": null,
              "width": "100px",
            }
          }
          toggleSort={[Function]}
        >
          <div
            className="rt-resizable-header-content"
          >
            Age
          </div>
          <Resizer
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          />
        </ThComponent>
      </TrComponent>
    </Thead>
    <Tbody
      className=""
      style={
        Object {
          "minWidth": "200px",
        }
      }
    >
      <TrGroupComponent
        key="0"
      >
        <TrComponent
          className="-odd"
        >
          <TdComponent
            className=""
            key="0-Name"
            onClick={[Function]}
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            John
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            onClick={[Function]}
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            23
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="1"
      >
        <TrComponent
          className="-even"
        >
          <TdComponent
            className=""
            key="0-Name"
            onClick={[Function]}
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Jane
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            onClick={[Function]}
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            26
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-0"
      >
        <TrComponent
          className="-padRow -odd"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-1"
      >
        <TrComponent
          className="-padRow -even"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-2"
      >
        <TrComponent
          className="-padRow -odd"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-3"
      >
        <TrComponent
          className="-padRow -even"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-4"
      >
        <TrComponent
          className="-padRow -odd"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-5"
      >
        <TrComponent
          className="-padRow -even"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-6"
      >
        <TrComponent
          className="-padRow -odd"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-7"
      >
        <TrComponent
          className="-padRow -even"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-8"
      >
        <TrComponent
          className="-padRow -odd"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-9"
      >
        <TrComponent
          className="-padRow -even"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-10"
      >
        <TrComponent
          className="-padRow -odd"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-11"
      >
        <TrComponent
          className="-padRow -even"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-12"
      >
        <TrComponent
          className="-padRow -odd"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-13"
      >
        <TrComponent
          className="-padRow -even"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-14"
      >
        <TrComponent
          className="-padRow -odd"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-15"
      >
        <TrComponent
          className="-padRow -even"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-16"
      >
        <TrComponent
          className="-padRow -odd"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
      <TrGroupComponent
        key="pad-17"
      >
        <TrComponent
          className="-padRow -even"
          style={Object {}}
        >
          <TdComponent
            className=""
            key="0-Name"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
          <TdComponent
            className=""
            key="1-Age"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <PadRowComponent />
          </TdComponent>
        </TrComponent>
      </TrGroupComponent>
    </Tbody>
  </TableComponent>
  <LoadingComponent
    loading={false}
    loadingText="Loading..."
  />
</div>
`;
